app/src/main/java/com/example/sharewhatyoucanproject/imgviewstyling/RoundRectCornerImageView.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.sharewhatyoucanproject.imgviewstyling
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Path
+import android.graphics.RectF
+import android.util.AttributeSet
+import com.bumptech.glide.Glide.init
+
+class RoundRectCornerImageView : androidx.appcompat.widget.AppCompatImageView {
```

Comment:
Both Coil and Glide contain options to transform images:

Coil: https://coil-kt.github.io/coil/transformations/
Glide: https://bumptech.github.io/glide/doc/transformations.html#default-transformations

In general using the library's built in tooling to transform the image will be more performant because it is cached after creation

Reply:
Thanks for the reference. Removed this class and directly used coil transformation.